### ISSUE TITLE
Unable to upgrade android

### ISSUE THREAD

**Issue by mark-prins** [msupply-foundation/open-msupply#4600]
## What went wrong? 😲

I had a version running yesterday and today built the rc3 version, deployed and received sadness:

![image](https://github.com/user-attachments/assets/7bfa7085-34d0-4161-a34b-9374d907fe2c)

Looking into why, I see that there's an issue with migration:

<img width="1018" alt="Screenshot 2024-08-16 at 11 36 44 AM" src="https://github.com/user-attachments/assets/67bd83d8-ca7d-4b07-8a95-1ea95479536a">

which is because there's nothing in the migration table
[android-data.zip](https://github.com/user-attachments/files/16631248/android-data.zip)

My question is this:
* how to diagnose these problems? ( for this one, I had to install a debug version and use android studio )
* can we offer a better fix experience? ( I've had to clear data and re-initialise, which means typing in all the details again )

## Suggested behaviour 🤔
* Could we display the last n log entries on this error screen?
* Have an option to re-initialise the database using the existing sync settings

Perhaps we could create a server method for the 'recreate database and initialise' function?
Possibly reading the log and displaying the text along with the `ErrorPage.java` static HTML?

## How to Reproduce 🔨

Steps to reproduce the behaviour:

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

**Comment by andreievg** [comment]
> Could we display the last n log entries on this error screen?
> Have an option to re-initialise the database using the existing sync settings

Agree to both, thanks @mark-prins

**Comment by andreievg** [comment]
Also we discussed this migration issue is one off

**Comment by jmbrunskill** [comment]
@aimee-mcneil-melville isn't this solved by your PR? You can see the logs now right?

**Comment by aimee-mcneil-melville** [comment]
@jmbrunskill yes viewing logs on the error page is in PR for 2.9 (not merged just yet) - issue #8081. 

Re-initialising hasn't been addressed though! So still to be fixed as part of this issue
